<a href="https://colab.research.google.com/github/Hari1331/Hari_DS/blob/master/Exploration.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:

import tensorflow as tf
print(tf.__version__)


1.14.0


Let's now look at the layers in your model
Computer Vision Example: Let's take a look at a scenario where we can recognize different items of clothing, trained from a dataset containing 10 different types.

TensorFlow Version : 1.14.0

Get the dataset from the Fashion MNIST

In [0]:
mnist = tf.keras.datasets.mnist

Load the Dataset from the Fashiin MNIST

In [3]:
(training_images, training_labels) ,  (test_images, test_labels) = mnist.load_data()

11493376/11490434 [==============================] - 0s 0us/step


Test and Train the images 255 * 255 pikcels


In [0]:
training_images = training_images/255.0
test_images = test_images/255.0

In [0]:
model = tf.keras.models.Sequential([tf.keras.layers.Flatten(),
                                    tf.keras.layers.Dense(1024, activation=tf.nn.relu),
                                    tf.keras.layers.Dense(10, activation=tf.nn.softmax)])

Compile the Categorical


In [0]:
model.compile(optimizer = 'adam',
              loss = 'sparse_categorical_crossentropy')

You do this by compiling it with an optimizer and loss function as before -- and then you train it by calling model.fit asking it to fit your training data to your training labels

In [17]:
model.fit(training_images, training_labels, epochs=5)

Epoch 1/5
60000/60000 [==============================] - 15s 249us/sample - loss: 0.1860
Epoch 2/5
60000/60000 [==============================] - 15s 251us/sample - loss: 0.0743
Epoch 3/5
60000/60000 [==============================] - 15s 248us/sample - loss: 0.0482
Epoch 4/5
60000/60000 [==============================] - 15s 249us/sample - loss: 0.0339
Epoch 5/5
60000/60000 [==============================] - 15s 246us/sample - loss: 0.0264


We can call model.evaluate, and pass in the two sets, and it will report back the loss for each.

In [8]:
model.evaluate(test_images, test_labels)

10000/10000 [==============================] - 1s 83us/sample - loss: 0.0768


0.07676124255598989

Returned a accuracy of about .0768, which means it was about 77% accurate.

In [0]:
classifications = model.predict(test_images)

Get the Classifications Results

In [10]:
print(classifications[0])

[2.6048478e-12 5.4919952e-10 3.3237857e-09 6.3537764e-10 2.9638419e-16
 5.8988562e-13 3.3978835e-14 1.0000000e+00 3.2376407e-11 6.6738182e-10]


Hint: try running print(test_labels[0]) -- and you'll get a 9. Does that help you understand why this list looks the way it does?

In [11]:
print(test_labels[0])

7
